### ISSUE TITLE
Add a stop button

### ISSUE THREAD

**Issue by barrucadu** [barrucadu/lainonlife#29]
Currently the pause button, when resumed, causes the player to continue where it left off.  It's now behind the stream.  Better behaviour would be to just resume from wherever the stream is up to.

**Comment by barrucadu** [comment]
This is a pretty small fix, *but* it does result in a few seconds of silence after resuming, while the stream buffers again.

```diff
diff --git a/frontend/js/player.js b/frontend/js/player.js
index ecb0ebf..a52807a 100644
--- a/frontend/js/player.js
+++ b/frontend/js/player.js
@@ -3,17 +3,19 @@ const LainPlayer = (() => {
     const audioTag = document.getElementById("audio");
     audioTag.volume = 0.51;
     let updateInterval;
+    let currentSrc = audioTag.currentSrc;

     function changeSource(source) {
         const paused = audioTag.paused;
         // Use either the ogg or mp3 stream, depending on what the current one is.
-        if(audioTag.currentSrc.endsWith("ogg")) {
-            audioTag.src = "/radio/" + source + ".ogg";
+        if(source.endsWith("ogg")) {
+            currentSrc = "/radio/" + source + ".ogg";
         } else {
-            audioTag.src = "/radio/" + source + ".mp3";
+            currentSrc = "/radio/" + source + ".mp3";
         }

         // Load the new audio stream.
+        audioTag.src = currentSrc;
         audioTag.load();

         // Start playing, if it was before.
@@ -35,9 +37,14 @@ const LainPlayer = (() => {

     function togglePlay() {
         if(audioTag.paused) {
+            audioTag.src = currentSrc;                                                                                                                                                                    +            audioTag.load();                                                                                                                                                                                           audioTag.play();                                                                                                                                                                                       } else {                                                                                                                                                                                                       audioTag.pause();                                                                                                                                                                             +            // clear the src and currentTime, to avoid buffering more streaming audio.                                                                                                                    +            audioTag.src = '';
+            audioTag.currentTime = 0;
         }
         updatePlayButton();
     }
```

Leaving it for now as I'm not sure this is better.

**Comment by barrucadu** [comment]
In addition to the current play/pause toggle, add a stop button which doesn't buffer.

**Comment by AlexFence** [comment]
should the stop button replace the pause button or should it be kept?

**Comment by barrucadu** [comment]
I think it should be next to it: `[stop] [play / pause toggle] [volume]`

**Comment by TypicalFence** [comment]
@barrucadu this can be closed as of #40
